app/assets/src/components/views/PipelineViz.jsx

```diff
@@ -0,0 +1,337 @@
+import React from "react";
+import { forEach } from "lodash";
+import PropTypes from "prop-types";
+import ReactPanZoom from "@ajainarayanan/react-pan-zoom";
+
+import RemoveIcon from "~/components/ui/icons/RemoveIcon";
+import NetworkGraph from "~/components/visualizations/NetworkGraph.js";
+import cs from "./pipeline_viz.scss";
+
+const START_NODE_ID = -1;
+const END_NODE_ID = -2;
+
+class PipelineViz extends React.Component {
+  constructor(props) {
+    super(props);
+    this.pipelineVersion = this.props.stageResults.pipeline_version;
+    this.stageGraphsInfo = [];
+    this.graphs = [];
+
+    this.stageNames = [
+      "Host Filtering",
+      "GSNAPL/RAPSEARCH alignment",
+      "Post Processing",
+    ];
+    if (this.props.admin) {
+      this.stageNames.push("Experimental");
+    }
+
+    this.state = {
+      stagesOpened: [true, true, true, true],
+      zoom: 1,
+    };
+
+    this.onMouseWheelZoom = this.onMouseWheelZoom.bind(this);
+  }
+
+  componentDidMount() {
+    this.drawGraphs();
+  }
+
+  onMouseWheelZoom(e) {
+    const zoomChange = (e.deltaY < 0 ? 1 : -1) * this.props.zoomChangeInterval;
+    this.setState({ zoom: this.state.zoom + zoomChange });
+  }
+
+  toggleStage(index) {
+    const updatedStagesOpened = [...this.state.stagesOpened];
+    updatedStagesOpened[index] = !updatedStagesOpened[index];
+    this.setState({ stagesOpened: updatedStagesOpened });
+  }
+
+  modifyStepNames() {
+    // Strips 'PipelineStep[Run/Generate]' from front of each step name.
+    // TODO(ezhong): Consider adding 'name' field to dag_json later.
+    forEach(this.props.stageResults.stages, (stageData, _) => {
+      stageData.steps.forEach(step => {
+        if (step.class.substring(0, 12) == "PipelineStep") {
+          step.class = step.class.substring(12);
+        }
+        if (step.class.substring(0, 3) == "Run") {
+          step.class = step.class.substring(3);
+        }
+        if (step.class.substring(0, 8) == "Generate") {
+          step.class = step.class.substring(8);
+        }
+      });
+    });
+  }
+
+  populateNodeAndEdgeData(index, nodeData, edgeData) {
```

Comment:
not a blocker but populateNodeAndEdgeData got quite long and could be split off into more functions

Reply:
Split into `populateIntraNodeAndEdgeData` and `populateInterEdgeData`